### PR TITLE
220005 Fix devmock for purchase order, fix line id generator for subscription

### PR DESCRIPTION
**Pull request by d3rky**

### CODE REVIEW

File: devmock/endpoints.py [hunk 1]
```diff
@@ -118,23 +119,23 @@ def complete_order(id: str, order: Order):
...
 def inquire_order(
     id: str,
-    template: Annotated[dict, Body()],
+    templateId: Annotated[str, Body()],
     parameters: Annotated[dict, Body()],
 ):
     order = load_order(id)
     order["parameters"] = parameters
-    order["template"] = template
+    order["templateId"] = template
```
Comment (ffaraone) [inline devmock/endpoints.py:138]: do the team have a bug for that ? According to docs is as it was before :)

File: devmock/mockgendata.py [hunk 2]
```diff
@@ -313,61 +316,79 @@ def gen_purchase_order(
...
-                gen_param("CompanyName"),
-                gen_param("PreferredLanguage"),
-                gen_param("Address"),
-                gen_param("Contact"),
+            "ordering": [
+                gen_param("CompanyName", "companyName"),
+                gen_param("PreferredLanguage", "preferredLanguage"),
+                gen_param("Address", "address"),
+                gen_param("Contact", "contact"),
+                gen_param("AgreementType", "agreementType", value="New"),
```
Comment (ffaraone) [inline devmock/mockgendata.py:363]: This is for migrations ?

**Review by ffaraone — CHANGES_REQUESTED** [review]

File: devmock/endpoints.py [hunk 3]
```diff
@@ -51,7 +52,7 @@ def list_orders(request: Request):
     query = unquote(request.scope.get("query_string", b"").decode())
     filter_instance = OrdersFilter()
     filtered_orders, count, limit, offset = filter_instance.apply(query, orders)
-    response["data"] = filtered_orders
+    response["data"] = orders
```
[the remark this answers is not in the document]
Comment (d3rky) [reply devmock/endpoints.py:55]: Yes! I did it for tests :-( reverting it back

File: devmock/endpoints.py (hunk 1 above)
Comment (d3rky) [reply devmock/endpoints.py:138]: I think yes, but that's the current implementation. If it was changed --> I'll revert it

File: devmock/mockgendata.py (hunk 2 above)
Comment (d3rky) [reply devmock/mockgendata.py:363]: Yes

**Review by ffaraone — APPROVED** [review]

**Pull request merged**